src/rules/fileNameCasingRule.ts

```diff
@@ -28,6 +28,74 @@ enum Casing {
     SnakeCase = "snake-case",
 }
 
+type RegexConfig = Record<string, Casing>;
+
+type SimpleConfig = Casing;
+
+type Config = SimpleConfig | RegexConfig;
+
+interface ValidatorSuccessfulResult {
+    valid: true;
+}
+
+interface ValidatorFailureResult {
+    valid: false;
+    failedCasing: Casing;
+}
+
+type ValidatorResult = ValidatorSuccessfulResult | ValidatorFailureResult;
+
+type Validator<T> = (casing: T) => (sourceFile: ts.SourceFile) => ValidatorResult;
+
+const rules = [Casing.CamelCase, Casing.PascalCase, Casing.KebabCase, Casing.SnakeCase];
+
+function isCorrectCasing(fileName: string, casing: Casing): boolean {
+    switch (casing) {
+        case Casing.CamelCase:
+            return isCamelCased(fileName);
+        case Casing.PascalCase:
+            return isPascalCased(fileName);
+        case Casing.KebabCase:
+            return isKebabCased(fileName);
+        case Casing.SnakeCase:
+            return isSnakeCased(fileName);
+    }
+}
+
+const RegexValidator: Validator<RegexConfig> = casingConfig => (sourceFile): ValidatorResult => {
```

Comment:
Naming convention for these would be pascalCase, like `regexValidator`.

Nit: consider also changing them to something like `validateWithRegex` and `validateWithSimpleConfig`. A bit more clear on what they're doing.